packages/react-native-web/src/exports/Image/index.js

```diff
@@ -94,56 +100,105 @@ function getFlatStyle(style, blurRadius, filterId) {
   return [flatStyle, resizeMode, _filter, tintColor];
 }
 
-function resolveAssetDimensions(source) {
-  if (typeof source === 'number') {
-    const { height, width } = getAssetByID(source);
-    return { height, width };
-  } else if (
-    source != null &&
-    !Array.isArray(source) &&
-    typeof source === 'object'
-  ) {
-    const { height, width } = source;
-    return { height, width };
-  }
+function resolveAssetDimensions(source: ImageSource) {
+  const { height, width } = source;
+  return { height, width };
 }
 
-function resolveAssetUri(source): ?string {
-  let uri = null;
+function resolveSource(source: ?Source): ImageSource {
+  let resolvedSource = { uri: '' };
+
   if (typeof source === 'number') {
-    // get the URI from the packager
-    const asset = getAssetByID(source);
-    let scale = asset.scales[0];
-    if (asset.scales.length > 1) {
-      const preferredScale = PixelRatio.get();
-      // Get the scale which is closest to the preferred scale
-      scale = asset.scales.reduce((prev, curr) =>
-        Math.abs(curr - preferredScale) < Math.abs(prev - preferredScale)
-          ? curr
-          : prev
+    resolvedSource = resolveNumericSource(source);
+  } else if (typeof source === 'string') {
+    resolvedSource = resolveStringSource(source);
+  } else if (Array.isArray(source)) {
+    if (process.env.NODE_ENV !== 'production') {
+      console.warn(
+        'The <Image> component does not support multiple sources passed as array, falling back to the first source in the list',
+        { source }
       );
     }
-    const scaleSuffix = scale !== 1 ? `@${scale}x` : '';
-    uri = asset
-      ? `${asset.httpServerLocation}/${asset.name}${scaleSuffix}.${asset.type}`
-      : '';
-  } else if (typeof source === 'string') {
-    uri = source;
+
+    return resolveSource(source[0]);
   } else if (source && typeof source.uri === 'string') {
-    uri = source.uri;
+    resolvedSource = resolveObjectSource(source);
   }
 
-  if (uri) {
-    const match = uri.match(svgDataUriPattern);
-    // inline SVG markup may contain characters (e.g., #, ") that need to be escaped
+  if (resolvedSource.uri) {
+    const match = resolvedSource.uri.match(svgDataUriPattern);
     if (match) {
-      const [, prefix, svg] = match;
-      const encodedSvg = encodeURIComponent(svg);
-      return `${prefix}${encodedSvg}`;
+      resolvedSource = resolveSvgDataUriSource(resolvedSource, match);
     }
   }
 
-  return uri;
+  return resolvedSource;
+}
+
+// get the URI from the packager
```

Comment:
```suggestion
// Get the URI from the packager
```
Please update all other comments to use consistent capitalization. Also, I think this should be a block comment and not an inline comment.